Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM node:6-slim
+
+RUN apt-get update && apt-get install libpng12-0
+
+WORKDIR /reactapp
+
+RUN mkdir -p /reactapp
+ADD package.json /reactapp/package.json
```

Comment:
It's also a good practice to [use COPY instead of ADD](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/add-or-copy).
And repeating `/reactapp` is unnecessary as well.
You could simply write:
`COPY package.json .` 
or `COPY . .` on line 10 bellow